src/decycle.ts

```diff
@@ -28,6 +28,7 @@ export const decycle = (obj: unknown, replacer?: (value: any) => any) => {
           curObj[name] = derez(value[name], `${path}/${name}`);
         });
       }
+      objs.delete(value);
```

Comment:
I think this might have a side-effect you didn't intend to have.
```ts
  it('should not deference twice', () => {
    const obj = {
      foo: {
        get c() {
          return obj.foo;
        },
      },
      c: {
        get a() {
          return obj.foo;
        },
      },
    };

    expect(decycle(obj)).toEqual({
      c: {
        a: {
          $ref: '#/foo',
        },
      },
      foo: {
        c: {
          $ref: '#/foo',
        },
      },
    });
  });
```
such a test fails.

![image](https://user-images.githubusercontent.com/9273484/117353890-00c09e80-aeb1-11eb-86d0-1733c2f14d85.png)

Reply:
or do we realllly want to keep such objects?
Seems totally redundant to me.

Reply:
Just going to put this here too for other peoples reference, I think your test might be incorrect.

```ts
it.only('should not deference twice', () => {
  const obj = {
    foo: {
      get c() {
        return obj.foo;
      },
    },
    c: {
      get a() {
        // wouldn't this return, and then c would be the actual circular 				javascript ref? Because the getter returns c() again?
        // {
        //	get c() {
        //	  return obj.foo;
        // },
        return obj.foo;
      },
    },
  };
​
  expect(decycle(obj)).toEqual({
    c: {
      a: {
        c: {
    	    $ref: '#/c/a',
        },
      },
    },
    foo: {
      c: {
        $ref: '#/foo',
      },
    },
  });
});
```